src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -81,43 +75,59 @@ public PowerShellKernel() : base(DefaultKernelName)
 
         private PowerShell CreatePowerShell()
         {
-            const string PSTelemetryEnvName = "POWERSHELL_DISTRIBUTION_CHANNEL";
-            const string PSTelemetryChannel = "dotnet-interactive-powershell";
-            const string PSModulePathEnvName = "PSModulePath";
-
             // Set the distribution channel so telemetry can be distinguished in PS7+ telemetry
             Environment.SetEnvironmentVariable(PSTelemetryEnvName, PSTelemetryChannel);
 
             // Create PowerShell instance
             var iss = InitialSessionState.CreateDefault2();
-            if(Platform.IsWindows)
+            if (Platform.IsWindows)
             {
                 // This sets the execution policy on Windows to RemoteSigned.
                 iss.ExecutionPolicy = ExecutionPolicy.RemoteSigned;
             }
 
             // Set $PROFILE.
-            PSObject profileValue = DollarProfileHelper.GetProfileValue();
+            var profileValue = DollarProfileHelper.GetProfileValue();
             iss.Variables.Add(new SessionStateVariableEntry("PROFILE", profileValue, "The $PROFILE."));
 
             var runspace = RunspaceFactory.CreateRunspace(_psHost, iss);
             runspace.Open();
             var pwsh = PowerShell.Create(runspace);
 
             // Add Modules directory that contains the helper modules
-            string psModulePath = Environment.GetEnvironmentVariable(PSModulePathEnvName);
             string psJupyterModulePath = Path.Join(
-                Path.GetDirectoryName(typeof(PowerShellKernel).Assembly.Location),
-                "Modules");
+               Path.GetDirectoryName(typeof(PowerShellKernel).Assembly.Location),
+               "Modules");
 
-            Environment.SetEnvironmentVariable(
-                PSModulePathEnvName,
-                $"{psJupyterModulePath}{Path.PathSeparator}{psModulePath}");
+            AddModulePath(psJupyterModulePath);
+
+            // Get the AddAccelerator method
+            var acceleratorType = typeof(PSObject).Assembly.GetType("System.Management.Automation.TypeAccelerators");
+            _addAccelerator = acceleratorType?.GetMethod("Add", new[] { typeof(string), typeof(Type) });
 
             RegisterForDisposal(pwsh);
             return pwsh;
         }
 
+        public void AddModulePath(string modulePath)
```

Comment:
Is this method (and `AddAccelerator`) used anywhere? I found `KernelExtension.fs` from the other PR has the same implementation.

Reply:
the other repo has the duplicated code, with this pr I am planning to move the api on the powershellkernel surface and use it from extensions

Reply:
sounds good